src/couch_replicator/src/couch_replicator_auth.erl

```diff
@@ -0,0 +1,100 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_replicator_auth).
+
+
+-export([
+    initialize/1,
+    update_headers/2,
+    handle_response/4,
+    cleanup/1
+]).
+
+
+-include("couch_replicator_api_wrap.hrl").
```

Comment:
Since we pass an #httpdb record to the `initialize/1` callback, should this file be moved to the `include` dir, and included using `-include_lib("couch_replicator/include/couch_replicator_api_wrap.hrl").`, so that other couch_replicator_auth implementations can access it as well?

Reply:
Good point, Jay. I'll update the PR.